src/backend/scripts/setup_localdata.py

```diff
@@ -63,7 +64,47 @@ def create_test_user(session, group, user_id, name):
     return u
 
 
-def create_sample(session, group, uploaded_by_user, suffix, is_failed=False):
+def create_location(session, region, country, division, location):
+    location = (
+        session.query(Location)
+        .filter(
+            and_(
+                Location.region == region,
+                Location.country == country,
+                Location.division == division,
+                Location.location == location,
+            )
+        )
+        .one_or_none()
+    )
+    if location:
+        print("Location already exists")
+        return location
+    print("Creating location")
+    location = Location(
+        region=region,
+        country=country,
+        division=division,
+        location=location,
+    )
+    session.add(location)
+    session.commit()
+    location = (
+        session.query(Location)
+        .filter(
+            and_(
+                Location.region == region,
+                Location.country == country,
+                Location.division == division,
+                Location.location == location,
+            )
+        )
+        .one_or_none()
```

Comment:
same question as above -- does SA not automatically populate the location.id column when it writes to the db?

Reply:
I tried several different approaches to debug this...and I'm mystified why a User gets an if upon creation but a Location only gets one when it's been committed.

Reply:
And same here -- this seems to work ok for me without the additional query.